delta/counter_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package delta_test
```

Comment:
Why not just make this local tests with `package delta`?

Reply:
I tend to write blackbox unit tests unless I have to just to avoid depending on internals too much. Are there more scenarios you think I should add?

Reply:
Yea, I guess this is reasonable to do as-a-library tests.